### DISCUSSION TITLE
How to install fonts to chiseled images

### DISCUSSION THREAD

**Discussion by tri-bao** [dotnet/dotnet-docker#5971] [Q&A]
I'm converting our `6.0.35-jammy` distro-based runtime images to distroless `6.0.35-jammy-chiseled` images. I need fonts in the runtime. In the distro-based image, they are installed as following

```
RUN DEBIAN_FRONTEND=noninteractive \
         echo "ttf-mscorefonts-installer msttcorefonts/accepted-mscorefonts-eula select true" | debconf-set-selections \
    && apt-get update \
    && apt-get install -y --no-install-recommends \
      libfontconfig1 \
      fontconfig \
      fonts-freefont-ttf \
      msttcorefonts 
```

I found `libfontconfig1` and `fontconfig` slices in `canonical/chisel-releases` and can cut the slices as following.

```
RUN mkdir -p /rootfs/var/lib/dpkg/ \
    && chisel-wrapper \
        --generate-dpkg-status /rootfs/var/lib/dpkg/status -- \
        --release ubuntu-22.04\
        --root /rootfs \
            libfontconfig1_libs \
            fontconfig-config_config

...
# copy the rootfs into the final chiseled image
```

But I can find ones for `fonts-freefont-ttf` and `msttcorefonts`. Any ideas how to install these font packages to the `6.0.35-jammy-chiseled` image? Thanks.

**Comment by lbussell** [comment]
If there are no existing slices for `fonts-freefont-ttf` and `msttcorefonts`, you can contribute them to the [canonical/chisel-releases](https://github.com/canonical/chisel-releases?tab=readme-ov-file#adding-new-slice-definitions) repo - see [Adding New Slice Definitions](https://github.com/canonical/chisel-releases?tab=readme-ov-file#adding-new-slice-definitions).

@cjdcordeiro, is there a way to add an entire non-slice package from the default Ubuntu repositories, without contributing it as a slice? For example, if a package has no need to be sliced?

**Reply by cjdcordeiro** [reply]
:wave: 

I'm afraid not. The only two options are:
 - use a custom local chisel-release, where you'd add the fonts packages as slice definitions files, but simply use a glob to refer to all font files (e.g. `/usr/share/fonts/truetype/**`)
 - propose it upstream, as @lbussell mentioned
 
 We have quite a few [examples](https://github.com/canonical/chisel-releases/blob/ubuntu-24.04/slices/fonts-ubuntu.yaml) of sliced font packages.

**Reply by tri-bao** [reply]
Thanks @lbussell and @cjdcordeiro I'll create a local slice for this.
